torch/csrc/jit/python/python_ivalue.h

```diff
@@ -11,17 +11,21 @@ namespace ivalue {
 struct C10_EXPORT ConcretePyObjectHolder final : PyObjectHolder {
  public:
   static c10::intrusive_ptr<PyObjectHolder> create(py::object py_obj) {
-    return c10::make_intrusive<ConcretePyObjectHolder>(py_obj);
+    return c10::make_intrusive<ConcretePyObjectHolder>(std::move(py_obj));
   }
 
   PyObject* getPyObject() override {
     return py_obj_.ptr();
   }
 
-  ~ConcretePyObjectHolder() {}
+  ~ConcretePyObjectHolder() {
+    pybind11::gil_scoped_acquire ag;
+    py_obj_ = py::none();
+  }
```

Comment:
I guess with this, we don't need the Future Deleter helper?

Reply:
Ah, right, we should be able to drop that lock.